### PR TITLE
Add: stirling control flags && Mod: dev base image

### PR DESCRIPTION
**Pull request by KayzzzZ**
* Add control flag for stirling
* Use GLOG for logs instead of using cout
* Update pixie module

Signed-off-by: qianlu.kk <qianlu.lk@alibaba-inc.com>

### CODE REVIEW

File: probe/scripts/run_docker.sh [hunk 1]
```diff
@@ -3,13 +3,12 @@ workspace_root=$(realpath "${script_dir}/../")
 
 # Docker image information.
 #docker_image_with_tag="kindlingproject/kindling-compiler"
-docker_image_with_tag="kindlingproject/kindling-compiler"
+# docker_image_with_tag="kindlingproject/kindling-compiler"
+docker_image_with_tag="kindlingproject/kindling-compiler:dev"
```
Comment (dxsup) [inline probe/scripts/run_docker.sh:7]: I think this is unnecessary because once your PR is merged, we will update the image using the tag `latest`.

**Review by dxsup — CHANGES_REQUESTED** [review]

**Comment by jundizhou** [comment]
We should point to this repo:
https://github.com/Kindling-project/pixie

**Review by dxsup — APPROVED** [review]

File: probe/src/probe/publisher/publisher.cpp [hunk 2]
```diff
@@ -245,9 +227,8 @@ px::Status publisher::consume_uprobe_data(uint64_t table_id, px::types::TabletID
         auto tinfo = m_inspector->get_thread_ref(pid, true, true, true);
         if (tinfo) {
             gevt.container_id = tinfo->m_container_id;
-            // std::cout << "[qianlu] find container_id for pid:" << pid << " container_id:" << gevt.container_id << std::endl;
         } else {
-            std::cout << "[qianlu] cannot find container_id for pid:" << pid << std::endl;
+		LOG(WARNING) << "[stirling] cannot find container_id for pid:" << pid << std::endl;
```
Comment (jundizhou) [inline probe/src/probe/publisher/publisher.cpp:231]: I think there will be performance problems here

File: probe/WORKSPACE [hunk 3]
```diff
@@ -4,7 +4,7 @@ load("@bazel_tools//tools/build_defs/repo:git.bzl","git_repository")
 git_repository(
     name = "px",
     remote = "https://github.com/KayzzzZ/pixie.git",
-    commit = "6b853b96987e570e0979620537c23df5850c7b76",
+    commit = "2d45f14dc69c060260c5634b7718f05f3f0c7bdb",
```
Comment (jundizhou) [inline probe/WORKSPACE:7]: Please change repo to kindling-project/pixie

File: probe/src/probe/converter/uprobe_converter.cpp [hunk 4]
```diff
@@ -48,6 +48,20 @@ void uprobe_converter::convert(void *evt) {
...
+  auto sa_attr = kevt->add_user_attributes();
+  sa_attr->set_key("source_addr");
+  auto sa_value = new AnyValue();
+  sa_value->set_string_value(gevt->source_addr);
+  sa_attr->set_allocated_value(sa_value);
+
+  // source_port
+  auto sp_attr = kevt->add_user_attributes();
+  sp_attr->set_key("source_port");
+  auto sp_value = new AnyValue();
```
Comment (jundizhou) [inline probe/src/probe/converter/uprobe_converter.cpp:61]: Would it be better to reuse anyvalue?

File: probe/src/probe/publisher/publisher.cpp (hunk 2 above)
Comment (KayzzzZ) [reply probe/src/probe/publisher/publisher.cpp:231]: You mean this block will be executed frequently?

File: probe/WORKSPACE (hunk 3 above)
Comment (KayzzzZ) [reply probe/WORKSPACE:7]: ok ...

File: probe/src/probe/converter/uprobe_converter.cpp (hunk 4 above)
Comment (KayzzzZ) [reply probe/src/probe/converter/uprobe_converter.cpp:61]: Protobuf will manage the object(AnyValue)'s lifestyle. Which means Protobuf will destroy the object but we don't know when and there is no callback... So I think we can't reuse the memory.

File: probe/src/probe/publisher/publisher.cpp (hunk 2 above)
Comment (dxsup) [reply probe/src/probe/publisher/publisher.cpp:231]: How frequently would it log?

Comment (KayzzzZ) [reply probe/src/probe/publisher/publisher.cpp:231]: It logs when we failed to find the corresponding `container_id` from `sinsp`.

Comment (jundizhou) [reply probe/src/probe/publisher/publisher.cpp:231]: I think it's normal for the virtual machine process to have a request. Is it necessary to print this line of log

**Review by jundizhou — APPROVED** [review]

**Pull request merged**